src/classes/HH_HouseholdNaming.cls

```diff
@@ -75,60 +78,72 @@ public without sharing class HH_HouseholdNaming {
+        Map<Id, List<Contact>> hhIdToContactsMap = getHhIDContactMap(hhids);
 
-        map<id, list<Contact>> hhIDContactMap = new map<id, list<Contact>>();
+        //loop through the households needing name changes, if necessary, make them.
+        for (SObject h : householdsNeedingUpdates) {
+            List<Contact> listCon = getContactsForHousehold(h, hhIdToContactsMap);
 
-        //sort contacts by household
-        for (Contact c : contactlist){
-            if(!hhIDContactMap.containskey(c.hhId__c)){
-                hhIDContactMap.put(c.hhId__c, new list<Contact>{c});
-            }
-            else{
-                list<Contact> clist = hhIDContactMap.get(c.hhId__c);
-                clist.add(c);
+            if (settings.isAdvancedHouseholdNaming()) {
+                new HouseholdName(h, listCon)
+                        .setNameFieldValuesOnHousehold();
             }
+
+            // now we also store the number of members on the Household!
+            h.put(UTIL_Namespace.StrTokenNSPrefix('Number_of_Household_Members__c'),
+                    listCon != null ? listCon.size() : 0);
         }
 
-        //loop through the households needing name changes, if necessary, make them.
-        for (SObject h : hhupdatelist){
-            ID hhId = string.valueOf(h.get('Id'));
-            list<Contact> listCon = hhIDContactMap.get(hhId);
+        if (!householdsNeedingUpdates.isEmpty()) {
+            // all or nothing, so errors not lost
+            UTIL_DMLService.updateRecords(householdsNeedingUpdates);
+        }
 
-            if (isAutomaticNaming()) {
-	            string customnameparam = '';
-	            if (h.get('npo02__SYSTEM_CUSTOM_NAMING__c') != null){
-	                customnameparam = string.valueOf(h.get('npo02__SYSTEM_CUSTOM_NAMING__c'));
-	            }
+        TDTM_ProcessControl.setRecursionFlag(TDTM_ProcessControl.flag.HH, false);
+    }
 
-	            if (!customnameparam.contains('Name'))
-	                h.put('Name', getHHName(listCon));
+    private Map<Id, List<Contact>> getHhIDContactMap(List<Id> hhids) {
+        return getHhIDContactMap(getHouseholdContacts(hhids));
+    }
 
-	            if (!customnameparam.contains('Formal_Greeting__c'))
-	                h.put('npo02__Formal_Greeting__c', getFormalName(listCon));
+    private List<SObject> getHouseholdsNeedingUpdates(List<Id> hhids) {
+        List<SObject> hhupdatelist = new List<SObject>();
+        hhupdatelist.addAll(getListHHObj(hhids));
+        hhupdatelist.addAll(getListHHAcct(hhids));
+        return hhupdatelist;
+    }
 
-	            if (!customnameparam.contains('Informal_Greeting__c'))
-	                h.put('npo02__Informal_Greeting__c', getInformalName(listCon));
-            }
+    private List<SObject> getListHHAcct(List<Id> hhids) {
+        List<SObject> listHHAcct = [select Id, Name, npo02__SYSTEM_CUSTOM_NAMING__c from Account where id IN :hhids];
```

Comment:
Suggestion - format the soql per our standard.

Reply:
👍🏽